src/viz/Ensight_Stream.cc

```diff
@@ -43,12 +44,13 @@ Ensight_Stream &endl(Ensight_Stream &s) {
  * \param file_name  Name of output file.
  * \param binary     If true, output binary.  Otherwise, output ascii.
  * \param geom_file  If true, then a geometry file will be dumped.
+ * \param decomposed If true, input is domain decomposed. Otherwise domain replicated.
  */
 Ensight_Stream::Ensight_Stream(const std::string &file_name, const bool binary,
-                               const bool geom_file)
-    : d_stream(), d_binary(false) {
+                               const bool geom_file, const bool decomposed)
+    : d_decomposed_stream(), d_serial_stream(), d_stream(), d_binary(false) {
```

Comment:
Am I the only one who thinks it's weird that we set d_binary to false in the initializer list, even though the binary parameter is passed to the constructor? It looks like we set d_binary in the open() call instead. (I know you didn't write that code, Matt... just something I noticed when you changed code in the vicinity :) )

Reply:
Yes, this is strange. I had intended to fix this and got lost in pointer world so I forgot... I'll fix it.

Reply:
I would seem that we should use a default argument value if the default value should be defaulted.
```c++
Ensight_Stream::Ensight_Stream(const std::string &file_name, const bool binary=faluse,
 const bool geom_file, const bool decomposed)
    : d_decomposed_stream(), d_serial_stream(), d_stream(), d_binary(binary) {
```

Reply:
So this is tricky. The set doesn't occur until open (which is a public call), which is only optionally called. Maybe I should ditch this constructor, and force the open function to be called independently. The way this is currently written is clunky. What do you all think?

Reply:
I think ofstream is a little like this (optional open in construction), though the full details I do not know for sure.

Reply:
Hmmm... Even though the setting of d_binary currently doesn't happen til the open() function, would it actually break other code to have it set in the constructor's initializer list? It seems like it wouldn't hurt to have it set for the Ensight_Stream object, even if the open() function isn't called on construction. The binary flag could be taken out of the open() arguments list, with the understanding that the open() function could only be called on the type of file the Ensight_Stream object was created for (binary or ascii).

Sorry if there is something I'm missing here! I'm also fine to leave it alone, since you didn't introduce it and thus shouldn't have to waste half your life trying to fix it :)

Reply:
I opted to set d_binary with binary, no additional default, because this was ignored anyway. I still don't love this constructor but I think it accomplishes the desired goal.